src/View/Form/ContextInterface.php

```diff
@@ -44,6 +44,15 @@ public function isPrimaryKey($field);
     public function isCreate();
 
     /**
+     * Sets the type of the form the context is created from.
+     *
+     * @param string $requestType The request type of the form the
+     *   context is created from.
+     * @return void
+     */
+    public function setRequestType($requestType);
```

Comment:
Looks okay to me, but I think the constructor would only be interested in whether the type is get or something else...
What do you think about ionas suggestion for being able to control the "data source order" (data then query, query then data, only data, only query) upon form creation? The default behavior could be the context depending solely on the form type (either data OR query) while this behavior could be alternated using a hashkey upon form creation that gets passed to the context...
I personally don't see the use case for it, but I would gladly implement it if you think functionality like this fits here...